clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -911,16 +913,16 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
      *
      * @param topics The list of topics to subscribe to
      * @throws IllegalArgumentException If topics is null or contains null or empty elements
+     * @throws IllegalStateException If {@code subscribe()} is called previously with pattern, or assign is called previously.
```

Comment:
"... is called previously... " without a subsequent call to `unsubscribe()`? Same below.